drivers/hmis/app/graphql/mutations/ce/submit_ce_referral_step.rb

```diff
@@ -11,21 +11,25 @@ class Ce::SubmitCeReferralStep < CleanBaseMutation
     argument :referral_id, ID, required: true
     argument :step_id, ID, required: true
     argument :input, Types::JsonObject, required: true
+    argument :form_definition_id, ID, required: true # The form that was used to submit the step
     argument :confirmed, Boolean, required: false
 
     field :step, Types::HmisSchema::CeReferralStep, null: true # nullable in case of errors
     field :referral, Types::HmisSchema::CeReferral, null: true # return the referral so the UI can respond appropriately if the referral's overall status has changed
 
-    def resolve(referral_id:, step_id:, input:, **_rest) # add 'confirmed' here later. For now, we ignore it
+    def resolve(referral_id:, step_id:, form_definition_id:, input:, **_rest) # add 'confirmed' here later. For now, we ignore it
       raise unless Hmis::Ce.configuration.enabled?
 
       referral = Hmis::Ce::Referral.viewable_by(current_user).find(referral_id)
       step = nil
       errors = HmisErrors::Errors.new
+      form_definition = Hmis::Form::Definition.find(form_definition_id)
+      raise unless form_definition.published?
```

Comment:
Should this be `raise if draft?` Since we're storing and resolving the form definition by id, I would expect its possible that a retired form may be used to "re"submit a step. (even if the UI doesn't allow it currently.)

Reply:
Yes sounds good, we can use the existing `valid_status_for_submit?` helper

Reply:
perfect!